src/autoload.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+spl_autoload_register(function ($className) {
+
+    // Stop here if class already loaded to avoid filesystem I/O.
+    if (class_exists($className, false)) {
+        return;
+    }
+
+    $dirs = [
+      'Client',
+      'Exception',
+      'Http',
+      'Result',
+      'Util',
+    ];
+
+    foreach ($dirs as $dir) {
```

Comment:
Again some example code but it was not using namespaces, so yes, I could extract the path from the namespace maybe, will try.

Reply:
When autoloading, the class that is being loaded is always writen using it's full name. Since we are only autoloading our own namespaced `BTCPayServer` classes, we can simply check if the class that is to be loaded starts with these characters. Easy.
Loading the file is as simple as replacing the `\` from the PHP class with `DIRECTORY_SEPARATOR` so it works OS independant.